compiler/src/dotty/tools/dotc/Driver.scala

```diff
@@ -40,10 +42,22 @@ class Driver extends DotClass {
 
   protected def sourcesRequired = true
 
+  /**
+   * Should the `ContextDocstrings` be set for this context? The `ContextDocstrings` is used
+   * to store doc comments unless `-Ydrop-comments` is set, or when TASTY is configured to
+   * unpickle the doc comments.
+   */
+  protected def shouldAddDocContext(implicit ctx: Context): Boolean = {
+    !ctx.settings.YdropComments.value || ctx.mode.is(Mode.ReadComments)
```

Comment:
Why do we have both a new mode and a compiler option?
Isn't `!ctx.settings.YdropComments <=> Mode.ReadComments`.

Reply:
Fair enough